.github/workflows/build_and_test.yml

```diff
@@ -105,6 +105,7 @@ jobs:
         if: github.ref == 'refs/heads/main'
         run: |
          echo "BAZEL_REMOTE_CACHE=--remote_upload_local_results=true \
+           --remote-timeout=10s \
```

Comment:
Should we instead put this in `.bazelrc`? These timeouts also suck when downloading from the cache while running local commands.

(Also, please add a comment explaining how this value of 10s was chosen. "10s was chosen arbitrarily as a not-too-long time" is a fine comment, for example! [it explains that we didn't do careful measurements to select this])